### PR TITLE
Merged changes from #79, #80, and #81

### LINKED ISSUE #74: Improve packing slip readability
- Maybe just use a whole page, or make it aware that the 2 labels will be side by side
- Bk packers suggested they want to get a label printer, but this is going to require some prototyping
- Include cluster id (populate on metadata for each ticket)
- Name + number 
- Reorder items on ticket based on how their grocery line is organized (bananas to the back)
- Separator lines between each item

### PR DESCRIPTION
**Pull request by leifwalsh**
I was starting to get confused about which changes were on which branches, so I merged all of mine into one PR.

closes #74

### CODE REVIEW

File: functions/airtable.js [hunk 1]
```diff
@@ -40,46 +45,85 @@ const TABLE_SCHEMAS = {
...
+ *
+ * @example
+ * // Gets bulk delivery routes for July 25th
+ * getRecordsWithFilter(BULK_DELIVERY_ROUTES_TABLE, { deliveryDate: new Date("2020-07-25") })
+ *
+ * @param {string} table The table to query.
+ * @param {Object.<string, any>} filter The filter to apply, keys should be columns defined in the schema for `table`.
+ * @returns {Promise<[string, Object, Object][]>} An array of airtable records.
+ */
+async function getRecordsWithFilter(table, filter) {
```
Comment (maybejustmaybe) [inline functions/airtable.js:104]: Nice, very helpful.

File: functions/scripts/check-bulk-order.js [hunk 2]
```diff
@@ -9,13 +10,23 @@ const {
 } = require('../airtable');
 
 async function main() {
+  // TODO: Do real argument parsing.
```
Comment (leifwalsh) [inline functions/scripts/check-bulk-order.js:13]: Oh, this file could still use some cleanup.

File: functions/airtable.js [hunk 3]
```diff
@@ -244,37 +294,265 @@ async function getBulkOrder(records) {
...
+      return [id, fields];
+    })
+  );
+
+  // WARNING: We cannot process orders in an async manner. We need them to be
+  // processed in a deterministic order, so that the accounting (which tickets
+  // need custom shoppers to buy bulk items we couldn't procure) will be the
+  // same between packing slips and shopping lists. Therefore, we need to be
+  // careful that we sort tickets the same way every time, and we do accounting
+  // for them sequentially.
```
Comment (maybejustmaybe) [inline functions/airtable.js:492]: I got the point from your comment; it took me a minute to process tho. An example could help make this a bit more comprehensible.

File: functions/templates/bulk-shopping-volunteer-email.md.mustache [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+   Sharpie or other pen/marker that won't smear or rub off.
+
+Please remember to bring at least one pen and some paper with you to the grocery
+store! We need drivers to drop these items off at the warehouse, **[221 Glenmore
+Ave, Gate 4]({{{warehouseMapsUrl}}}) by 12pm**, so please plan your shopping and a
+meetup time with your car teammate accordingly.
+
+Submit the [reimbursement form](https://airtable.com/shrvHf4k5lRo0I8F4) in the
+usual way, with the total amount you spent and images of your receipt(s). Just
+pick one ticket ID from your list and enter that. Please add a note that your
```
Comment (maybejustmaybe) [inline functions/templates/bulk-shopping-volunteer-email.md.mustache:24]: "pick one ticket ID from your list and enter that": is there any way we can improve this flow?

File: functions/airtable.js [hunk 5]
```diff
@@ -90,18 +134,16 @@ async function getRecordsWithStatus(table, status) {
 async function getChangedRecords(table, includeNullStatus = false) {
```
[the remark this answers is not in the document]
Comment (leifwalsh) [reply functions/airtable.js:134]: #85

File: functions/airtable.js [hunk 6]
```diff
@@ -244,37 +294,265 @@ async function getBulkOrder(records) {
...
+  }
+}
+
+/**
+ * Reconcile bulk delivery orders with procured items from Bulk Order table.
+ * @param {[string, Object, Object][]} allRoutes All bulk delivery routes for this week.
+ * @param {Date} deliveryDate Date these orders will go out.
+ * @returns {Promise<ReconciledOrder[]>} List of reconciled orders.
+ */
+async function reconcileOrders(allRoutes, deliveryDate) {
```
[the remark this answers is not in the document]
Comment (leifwalsh) [reply functions/airtable.js:462]: I made it an optional parameter. `email-bulk-shopping-volunteers.js` uses `allRoutes` for other things and it's a minor optimization/consistency thing but I'd like to only query for it once. `generate-packing-slips.js` gets simpler though.

File: functions/airtable.js (hunk 3 above)
Comment (leifwalsh) [reply functions/airtable.js:492]: Added one!

File: functions/templates/bulk-shopping-volunteer-email.md.mustache (hunk 4 above)
Comment (leifwalsh) [reply functions/templates/bulk-shopping-volunteer-email.md.mustache:24]: Filed #86

**Pull request merged**